Fix rendering issue of ruler numbers

PyvesB opened the pull request:
Fixes  #87, or rather the ruler number part of it.

This fix came to light after further digging into what was done in https://git.eclipse.org/r/#/c/161336/

Screenshot of it in action:
![Screenshot (188)](https://user-images.githubusercontent.com/10694593/82023115-0ca6c580-968e-11ea-8220-48f083fd690b.png)

Two questions came to mind before submitting this pull request:
* should the same approach be applied to the vertical bar as well?
* should the same approach be applied to all adapted widgets, not just StyledText?

Given that I had no concrete example in mind that would benefit from either of the above, I decided to keep on the safe side and limit the scope of this fix.

PyvesB commented on com.codeaffine.eclipse.swt/src/com/codeaffine/eclipse/swt/widget/scrollable/StyledTextAdapter.java:
```diff
@@ -313,6 +313,15 @@ public void setFont( Font font ) {
+    // widgets (e.g. ruler, see https://github.com/fappel/xiliary/issues/87).
+    // External code trying to unhide it would trigger a PaintEvent, hence the
+    // listener to ensure it's always hidden.
+    styledText.addPaintListener( e -> {
```
I'm expecting the performance impact of this to be negligible: underneath the hood, `setVisible` is a fast return if the value is already false.

fappel commented:
Thank you very much for your effort. I will merge this and have a look at the Test failure by the end of next week.

PyvesB commented:
I hadn't noticed the test failure, it's the new one I added which was passing fine locally. It may well be a timing issue, I added an extra `waitForReconciliation` after the initial `openShellWithoutLayout`, like some of the other tests have. However, the build failed because one of the project's dependencies could not be downloaded.

fappel commented:
Hi, I've merged the pull request and fixed the test problem. FlatScrollBarAdapter related functionality is MS Windows specific. Hence, a lot of according tests will break on other platforms. Adding the @ConditionalIgnore annotation ensures that the trouble making test
will only be executed on MS Windows. Unfortunately, there is no explicit build run for Windows System on Travis Xiliary build. So, we have to rely on execution of the test suite on local machines :-( Testing the solution, I noticed that the ruler number problems was still present on initial opening of an editor. I've changed the code slightly that the initial situation is taken care of. See commit 1623824 for details. The fix has already been published and your contribution has been noted in the release notes as last time. Thank you very much!
